Expensify/App
# fix inline code blocks for native platforms

Please review @sketchydroide.

### Details

#### Problem

We (Including other contributors) tried using a combination of Text & View tag to achieve the above UI but there are a couple of issues as follows:
 1. Text is rendered as a block box but we want inline text.
 2. In Another way, we can get the inline text but border and border-radius are not being rendered.

#### How I fixed it?
I had an idea of tokenizing the text. Then apply the style selectively. This is working well. I have made a small snap here.

1. we create one wrapper component which divides the text into tokens(words) divided based on word break. This component put the text tokens inline into a view and renders it.
2. We can then pass styles selectively to tokens and manage the border-radius for the first and last tokens.


### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes #986

### Tests / QA Steps

- Type a sentence with code block formatting (single ticks) in the middle, like this:
   This is a test to see `how code renders in the middle` of a sentence on mobile.
- Notice that it should render inline like normal text.

### Issue(Not blocker)
1. ~Inline Rendered code block is not vertically center with respect to the text.~ :heavy_check_mark: *Fixed*

### Tested On


### Screenshots


#### Web / Desktop

![localhost_8080_r_68438375](https://user-images.githubusercontent.com/24370807/115467018-7cceab80-a24e-11eb-8469-c5493ca823d0.png)

#### Android

<img width="300" src="https://user-images.githubusercontent.com/24370807/115466438-beab2200-a24d-11eb-8d61-56db7956a563.jpg" >




#### IOS
I don't have an IOS device thus I mimicked the logic [here](https://snack.expo.io/@rajat_parashar/text-wrapping) and which can be seen working well.

![image](https://user-images.githubusercontent.com/24370807/115466577-e8644900-a24d-11eb-9199-55b9afb7cf31.png)
